datafusion/core/src/config.rs

```diff
@@ -0,0 +1,187 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
If this module is  in `datafusion-core` it will not be able to be used in other modules directly. That seems ok for now 🤔  but I haven't fully thought about it